Content.Client/Lobby/UI/Roles/RequirementsSelector.xaml.cs

```diff
@@ -107,13 +106,55 @@ public RequirementsSelector()
 
     public void LockRequirements(FormattedMessage requirements)
     {
+        var requirementsLabel = new Label()
+        {
+            Text = Loc.GetString("role-timer-locked"),
+            Visible = true,
+            HorizontalAlignment = HAlignment.Center,
+            StyleClasses = {StyleBase.StyleClassLabelSubText},
+        };
+
+        _lockStripe.Children.Clear();
+        _lockStripe.AddChild(requirementsLabel);
+
         var tooltip = new Tooltip();
         tooltip.SetMessage(requirements);
         _lockStripe.TooltipSupplier = _ => tooltip;
         _lockStripe.Visible = true;
         _options.Visible = false;
     }
 
+    public void LockDueToBan(string banReason, DateTime? expirationTime)
+    {
+        var requirementsLabel = new Label()
+        {
+            Text = Loc.GetString("role-banned-locked"),
+            Visible = true,
+            HorizontalAlignment = HAlignment.Center,
+            FontColorOverride = Color.Red,
+            StyleClasses = {StyleBase.StyleClassLabelSubText},
+        };
+
+        _lockStripe.Children.Clear();
+        _lockStripe.AddChild(requirementsLabel);
+
+        var tooltip = new Tooltip();
+
+        var message = new FormattedMessage();
+        message.AddText(Loc.GetString("role-banned-reason", ("reason", banReason)));
+        message.AddText("\n");
```

Comment:
Pretty sure FormattedMessage has a PushText method or something that also handles newlines.

Reply:
There is one PushMarkup?
But it looks like it is obsoleted.
`    [Obsolete]
    public void PushMarkup(string markup)
    {
        AddMarkup(markup);
        PushNewline();
    }`
    
Changed the 145 line to be `message.PushNewline();`